packages/grid/x-grid-modules/src/components/checkbox-renderer.tsx

```diff
@@ -28,16 +28,14 @@ export const HeaderCheckbox: React.FC<ColParams> = React.memo(({ api }) => {
   }, [api, selectionChange]);
 
   return (
-    <div>
-      <Checkbox
-        indeterminate={isIndeterminate}
-        checked={isChecked}
-        onChange={handleChange}
-        className="MuiDataGrid-checkboxInput"
-        color="primary"
-        inputProps={{ 'aria-label': 'Select All Rows checkbox' }}
-      />
-    </div>
+    <Checkbox
```

Comment:
lighter DOM structure. Made possible with display flex, align center, justify center. I assume the div was added to fix the non-circular checkbox.